### ISSUE TITLE
Call to user defined addr proc results in compilation error when defined in a separate module under some weird esoteric circumstances

### ISSUE THREAD

**Issue by johnnovak** [nim-lang/Nim#4007]
Observe the following code that compiles just fine. Now, if you moved the first few lines that contain an ``addr`` proc definition into a separate file and then imported it, a compilation error would be incorrectly raised.

What's weird is that only the ``addr`` proc seems to be affected; other proc calls are compiling fine. Also, there are some "workarounds" (= more weirdness) to "bypass" this compiler error (see below after the example code).

```
type Vec4[T] = array[4, T]

proc vec4*[T](x, y, z, w: T): Vec4[T] =
  Vec4([x, y, z, w])

proc `addr`*[T](v: var Vec4[T]): ptr T =
  v[0].addr

proc stuff*[T](v: var Vec4[T]) = discard
# ---> SNIP <---  moving the above code into a separate file and then
# importing it will result in a compile error on the call to addr (last line)

proc someFun(v: ptr float) = discard

var
  modelDiffuse = vec4[float](1.0, 0.8, 0.8, 1.0)

modelDiffuse.stuff
someFun(modelDiffuse.addr)
```

But wait, it only gets better from here! If you then imported _another_ module, the code would compile fine again... but only if the two imports are in a _particular order_ (yes...) and if the other imported file has  a _certain minimum length_!

I must say I have no idea what's going on here, but I have put together some test cases to illustrate the various incarnations of the issue:

https://github.com/johnnovak/nim-lang-bugs/tree/master/import-addr

Tested with Nim 0.13.0 on OS X (64-bit).

**Comment by Araq** [comment]
You must not overload ``addr``. I think the spec says so too.

**Comment by johnnovak** [comment]
Ah, didn't know about that. Actually, I have encountered this issue when using the nim-glm library that overloads `addr` (by using macros).

But shouldn't the compiler raise an error then when an `addr` overload attempt is detected?

**Comment by johnnovak** [comment]
Btw, just searched for all occurrences of `addr` in the 0.13.0 manual and nothing is mentioned about overloading.

**Comment by Varriount** [comment]
Of course, I wonder if addr could be pseudo-overwritten via term-rewriting macros
